diff -ur jack-1.9.9.5/common/memops.c jack-1.9.9.5.new/common/memops.c
--- jack-1.9.9.5/common/memops.c	2012-12-10 10:26:56.000000000 +0100
+++ jack-1.9.9.5.new/common/memops.c	2014-05-29 22:09:16.698704719 +0200
@@ -445,12 +445,12 @@
 		_mm_store_ss((float*)z+1, (__m128)shuffled1);
 		_mm_store_ss((float*)z+2, (__m128)shuffled2);
 		_mm_store_ss((float*)z+3, (__m128)shuffled3);
+#endif
 
 		for (i = 0; i != 4; ++i) {
 			memcpy (dst, z+i, 3);
 			dst += dst_skip;
 		}
-#endif
 
 		nsamples -= 4;
 		src += 4;
