Index: libjack/thread.c
===================================================================
--- libjack/thread.c	(revision 4184)
+++ libjack/thread.c	(working copy)
@@ -99,6 +99,17 @@
 #endif /* USE_CAPABILITIES */
 }	
 
+static void
+jack_thread_touch_stack ()
+{
+	char buf[JACK_THREAD_STACK_TOUCH];
+	int i;
+
+	for (i = 0; i < JACK_THREAD_STACK_TOUCH; i++) {
+		buf[i] = (char) (i & 0xff);
+	}
+}
+
 static void*
 jack_thread_proxy (void* varg)
 {
@@ -107,13 +118,8 @@
 	void* warg;
 	jack_client_t* client = arg->client;
 
-	char buf[JACK_THREAD_STACK_TOUCH];
-	int i;
+	jack_thread_touch_stack ();
 
-	for (i = 0; i < JACK_THREAD_STACK_TOUCH; i++) {
-                buf[i] = (char) (i & 0xff);
-	}
-
 	if (arg->realtime) {
 		maybe_get_capabilities (client);
 		jack_acquire_real_time_scheduling (pthread_self(), arg->priority);

