diff --git a/common/JackControlAPI.cpp b/common/JackControlAPI.cpp
index a2e94fe..bda6dfe 100644
--- a/common/JackControlAPI.cpp
+++ b/common/JackControlAPI.cpp
@@ -1009,7 +1009,7 @@ jackctl_server_open(
         }
 
         paramlist = jackctl_create_param_list(driver_ptr->parameters);
-        if (paramlist == NULL) goto fail_delete;
+        //if (paramlist == NULL) goto fail_delete;
         rc = server_ptr->engine->Open(driver_ptr->desc_ptr, paramlist);
         jackctl_destroy_param_list(paramlist);
         if (rc < 0)
@@ -1288,7 +1288,7 @@ SERVER_EXPORT bool jackctl_server_load_internal(
     int status;
     if (server_ptr->engine != NULL) {
         JSList * paramlist = jackctl_create_param_list(internal->parameters);
-        if (paramlist == NULL) return false;
+        //if (paramlist == NULL) return false;
         server_ptr->engine->InternalClientLoad2(internal->desc_ptr->name, internal->desc_ptr->name, paramlist, JackNullOption, &internal->refnum, -1, &status);
         jackctl_destroy_param_list(paramlist);
         return (internal->refnum > 0);
@@ -1364,7 +1364,7 @@ SERVER_EXPORT bool jackctl_server_switch_master(jackctl_server * server_ptr, jac
 {
     if (server_ptr && server_ptr->engine) {
         JSList * paramlist = jackctl_create_param_list(driver_ptr->parameters);
-        if (paramlist == NULL) return false;
+        //if (paramlist == NULL) return false;
         jackctl_destroy_param_list(paramlist);
         bool ret = (server_ptr->engine->SwitchMaster(driver_ptr->desc_ptr, paramlist) == 0);
         jackctl_destroy_param_list(paramlist);
