commit 41a9e424372de810193389709f03a6e7e4049bc0
Author: Adrian Knoth <adi@drcomp.erfurt.thur.de>
Date:   Fri Jul 29 03:50:58 2011 +0200

    man: Fix typo in jack_disconnect manpage

diff --git a/man/jack_connect.0 b/man/jack_connect.0
index 602ac39..5ee8f8b 100644
--- a/man/jack_connect.0
+++ b/man/jack_connect.0
@@ -5,7 +5,7 @@
 \fB jack_connect\fR [ \fI-s\fR | \fI--server servername\fR ] [\fI-h\fR | \fI--help\fR ] port1 port2
 \fB jack_disconnect\fR [ \fI-s\fR | \fI--server servername\fR ] [\fI-h\fR | \fI--help\fR ] port1 port2
 .SH DESCRIPTION
-\fBjack_connect\fR connects the two named ports. \fBjack_connect\fR disconnects the two named ports.
+\fBjack_connect\fR connects the two named ports. \fBjack_disconnect\fR disconnects the two named ports.
 .SH RETURNS
 The exit status is zero if successful, 1 otherwise
 
