Index: linux/alsa/JackAlsaDriver.cpp
===================================================================
--- linux/alsa/JackAlsaDriver.cpp	(revisione 4125)
+++ linux/alsa/JackAlsaDriver.cpp	(copia locale)
@@ -1729,7 +1729,7 @@
                                         offset, contiguous)) < 0) {
             jack_error ("ALSA: could not complete playback of %"
             		PRIu32 " frames: error = %d", contiguous, err);
-            if (err != EPIPE && err != ESTRPIPE)
+            if (err != -EPIPE && err != -ESTRPIPE)
                 return -1;
         }
 
Index: linux/alsa/usx2y.c
===================================================================
--- linux/alsa/usx2y.c	(revisione 4125)
+++ linux/alsa/usx2y.c	(copia locale)
@@ -622,7 +622,7 @@
 					offset, nframes_)) < 0) {
 		jack_error ("ALSA/USX2Y: could not complete playback of %"
 			    PRIu32 " frames: error = %d", nframes_, err);
-		if (err != EPIPE && err != ESTRPIPE)
+		if (err != -EPIPE && err != -ESTRPIPE)
 			return -1;
 	}
 
