diff --git a/example-clients/wscript b/example-clients/wscript
index 27a687d..49dd7e2 100644
--- a/example-clients/wscript
+++ b/example-clients/wscript
@@ -137,14 +137,14 @@ def build(bld):
         prog.includes = os_incdir + ['../common/jack', '../common']
         prog.source = ['alsa_in.c', '../common/memops.c']
         prog.env.append_value("CFLAGS", "-DNO_JACK_ERROR")
-        prog.use = ['clientlib', 'ALSA', 'SAMPLERATE']
+        prog.use = ['clientlib', 'ALSA', 'SAMPLERATE', 'M']
         prog.target = 'alsa_in'
 
         prog = bld(features = 'c cprogram')
         prog.includes = os_incdir + ['../common/jack', '../common']
         prog.source = ['alsa_out.c', '../common/memops.c']
         prog.env.append_value("CFLAGS", "-DNO_JACK_ERROR")
-        prog.use = ['clientlib', 'ALSA', 'SAMPLERATE']
+        prog.use = ['clientlib', 'ALSA', 'SAMPLERATE', 'M']
         prog.target = 'alsa_out'
 
     for example_lib, example_lib_source in list(example_libs.items()):
