diff --git a/common/jack/systemdeps.h b/common/jack/systemdeps.h
index b104f2a..d3e799a 100644
--- a/common/jack/systemdeps.h
+++ b/common/jack/systemdeps.h
@@ -37,7 +37,7 @@ Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
             #define JACK_ALIGN(N) __declspec(align(N))
 
             /* A utility for declaring a type which is aligned to its size. */
-            #define JACK_ALIGNED_TYPE(T) JACK_ALIGN(sizeof(T)) T
+            #define JACK_ALIGNED_TYPE(T) JACK_ALIGN(8) T
 
         #else
 
